File: app/src/main/java/com/kumi/Bahamut/Pages/MailBoxPage_ItemView.java
package com.kumi.Bahamut.Pages;

import android.annotation.SuppressLint;
import android.content.Context;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;
import com.kumi.Bahamut.Pages.Model.MailBoxPageItem;

public class MailBoxPage_ItemView extends LinearLayout {
  private static int _count = 0;
  
  private TextView _author = null;
  
  private TextView _date = null;
  
  private View _divider_bottom = null;
  
  private TextView _mark = null;
  
  private TextView _number = null;
  
  private TextView _reply = null;
  
  private TextView _status = null;
  
  private TextView _title = null;
  
  public MailBoxPage_ItemView(Context paramContext) {
    super(paramContext);
    init();
  }
  
  public MailBoxPage_ItemView(Context paramContext, AttributeSet paramAttributeSet) {
    super(paramContext, paramAttributeSet);
    init();
  }
  
  private void init() {
    ((LayoutInflater)getContext().getSystemService("layout_inflater")).inflate(2131361845, (ViewGroup)this);
    this._status = (TextView)findViewById(2131230875);
    this._title = (TextView)findViewById(2131230876);
    this._number = (TextView)findViewById(2131230873);
    this._date = (TextView)findViewById(2131230870);
    this._mark = (TextView)findViewById(2131230872);
    this._author = (TextView)findViewById(2131230869);
    this._reply = (TextView)findViewById(2131230874);
    this._divider_bottom = findViewById(2131230871);
  }
  
  public void clear() {
    setTitle((String)null);
    setDate((String)null);
    setAuthor((String)null);
    setIndex(0);
    setRead(true);
    setReply(false);
    setMark(false);
  }
  
  protected void finalize() throws Throwable {
    super.finalize();
  }
  
  public void setAuthor(String paramString) {
    if (this._author != null) {
      if (paramString != null) {
        this._author.setText(paramString);
        return;
      } 
      this._author.setText("讀取中");
    } 
  }
  
  public void setDate(String paramString) {
    if (this._date != null) {
      if (paramString != null) {
        this._date.setText(paramString);
        return;
      } 
      this._date.setText("讀取中");
    } 
  }
  
  public void setDividerBottomVisible(boolean paramBoolean) {
    if (this._divider_bottom != null) {
      if (paramBoolean) {
        if (this._divider_bottom.getVisibility() != 0)
          this._divider_bottom.setVisibility(0); 
        return;
      } 
    } else {
      return;
    } 
    if (this._divider_bottom.getVisibility() != 8)
      this._divider_bottom.setVisibility(8); 
  }
  
  @SuppressLint({"DefaultLocale"})
  public void setIndex(int paramInt) {
    if (this._number != null) {
      if (paramInt > 0) {
        String str = String.format("%1$05d", new Object[] { Integer.valueOf(paramInt) });
        this._number.setText(str);
        return;
      } 
      this._number.setText("讀取中");
    } 
  }
  
  public void setItem(MailBoxPageItem paramMailBoxPageItem) {
    if (paramMailBoxPageItem != null) {
      setTitle(paramMailBoxPageItem.Title);
      setIndex(paramMailBoxPageItem.Number);
      setDate(paramMailBoxPageItem.Date);
      setAuthor(paramMailBoxPageItem.Author);
      setReply(paramMailBoxPageItem.isReply);
      setRead(paramMailBoxPageItem.isRead);
      setMark(paramMailBoxPageItem.isMarked);
      return;
    } 
    clear();
  }
  
  public void setMark(boolean paramBoolean) {
    if (paramBoolean) {
      this._mark.setVisibility(0);
      return;
    } 
    this._mark.setVisibility(4);
  }
  
  public void setRead(boolean paramBoolean) {
    if (paramBoolean) {
      this._status.setText("◇");
      this._title.setTextColor(-8355712);
      return;
    } 
    this._status.setText("◆");
    this._title.setTextColor(-1);
  }
  
  public void setReply(boolean paramBoolean) {
    if (paramBoolean) {
      this._reply.setVisibility(0);
      return;
    } 
    this._reply.setVisibility(4);
  }
  
  public void setTitle(String paramString) {
    if (this._title != null) {
      if (paramString != null) {
        this._title.setText(paramString);
        return;
      } 
      this._title.setText("讀取中...");
    } 
  }
}


/* Location:              C:\Users\kodak\Downloads\反編譯\dex-tools-v2.4\classes-dex2jar.jar!\com\kumi\Bahamut\Pages\MailBoxPage_ItemView.class
 * Java compiler version: 6 (50.0)
 * JD-Core Version:       1.1.3
 */

File: app/src/main/java/com/kota/Bahamut/pages/mailPage/MailBoxPageItemView.kt
package com.kota.Bahamut.pages.mailPage

import android.annotation.SuppressLint
import android.content.Context
import android.util.AttributeSet
import android.view.View
import android.widget.LinearLayout
import android.widget.TextView
import com.kota.Bahamut.R
import com.kota.Bahamut.pages.model.MailBoxPageItem
import com.kota.Bahamut.service.CommonFunctions.getContextString
import com.kota.Bahamut.service.CommonFunctions.getThemeColor
import java.util.Objects

class MailBoxPageItemView : LinearLayout {
    var authorTextView: TextView? = null
    var dateTextView: TextView? = null
    var dividerBottom: View? = null
    var markTextView: TextView? = null
    var numberTextView: TextView? = null
    var replyTextView: TextView? = null
    var statusTextView: TextView? = null
    var titleTextView: TextView? = null

    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs) {
        init()
    }

    constructor(context: Context?) : super(context) {
        init()
    }

    fun setItem(aItem: MailBoxPageItem?) {
        if (aItem != null) {
            setTitle(aItem.title)
            setIndex(aItem.itemNumber)
            setDate(aItem.date)
            setAuthor(aItem.author)
            setReply(aItem.isReply)
            setRead(aItem.isRead)
            setMark(aItem.isMarked)
            return
        }
        clear()
    }

    fun init() {
        val density = context.resources.displayMetrics.density
        fun dp(value: Float): Int = (value * density + 0.5f).toInt()

        orientation = VERTICAL
        layoutParams = LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.WRAP_CONTENT)

        val row = LinearLayout(context).apply {
            orientation = HORIZONTAL
            layoutParams = LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.WRAP_CONTENT)
        }

        val bgView = LinearLayout(context).apply {
            orientation = VERTICAL
            layoutParams = LayoutParams(0, LayoutParams.WRAP_CONTENT, 1f)
            setBackgroundColor(getThemeColor(R.attr.bahamut_pageBackground))
            setPadding(dp(8f), dp(4f), dp(8f), dp(4f))
        }

        val titleRow = LinearLayout(context).apply {
            orientation = HORIZONTAL
            layoutParams = LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.WRAP_CONTENT)
        }

        statusTextView = com.kota.telnetUI.textView.TelnetTextViewNormal(context).apply {
            id = R.id.MailBoxPage_ItemView_Status
            layoutParams = LayoutParams(dp(30f), LayoutParams.WRAP_CONTENT)
            setTextColor(getThemeColor(R.attr.bahamut_boardItemStatusColor))
        }
        titleRow.addView(statusTextView)

        titleTextView = com.kota.telnetUI.textView.TelnetTextViewNormal(context).apply {
            id = R.id.MailBoxPage_ItemView_Title
            layoutParams = LayoutParams(0, LayoutParams.WRAP_CONTENT, 1f)
            ellipsize = android.text.TextUtils.TruncateAt.END
            isSingleLine = true
            setTextColor(getThemeColor(R.attr.bahamut_boardItemNormalColor))
        }
        titleRow.addView(titleTextView)
        bgView.addView(titleRow)

        val metaRow = LinearLayout(context).apply {
            orientation = HORIZONTAL
            layoutParams = LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.WRAP_CONTENT)
        }

        numberTextView = com.kota.telnetUI.textView.TelnetTextViewSmall(context).apply {
            id = R.id.MailBoxPage_ItemView_Number
            layoutParams = LayoutParams(LayoutParams.WRAP_CONTENT, LayoutParams.WRAP_CONTENT)
            setTextColor(getThemeColor(R.attr.bahamut_boardItemNumberColor))
        }
        metaRow.addView(numberTextView)

        markTextView = com.kota.telnetUI.textView.TelnetTextViewSmall(context).apply {
            id = R.id.MailBoxPage_ItemView_mark
            layoutParams = LayoutParams(LayoutParams.WRAP_CONTENT, LayoutParams.WRAP_CONTENT).apply {
                marginStart = dp(12f)
            }
            text = getContextString(R.string.word_m)
            setTextColor(getThemeColor(R.attr.bahamut_boardItemMarkColor))
        }
        metaRow.addView(markTextView)

        replyTextView = com.kota.telnetUI.textView.TelnetTextViewSmall(context).apply {
            id = R.id.MailBoxPage_ItemView_Reply
            layoutParams = LayoutParams(LayoutParams.WRAP_CONTENT, LayoutParams.WRAP_CONTENT).apply {
                marginStart = dp(12f)
            }
            text = getContextString(R.string.word_r)
            setTextColor(getThemeColor(R.attr.bahamut_mailItemReplyColor))
        }
        metaRow.addView(replyTextView)

        dateTextView = com.kota.telnetUI.textView.TelnetTextViewSmall(context).apply {
            id = R.id.MailBoxPage_ItemView_Date
            layoutParams = LayoutParams(LayoutParams.WRAP_CONTENT, LayoutParams.WRAP_CONTENT).apply {
                marginStart = dp(12f)
            }
            setTextColor(getThemeColor(R.attr.bahamut_boardItemDateColor))
        }
        metaRow.addView(dateTextView)

        authorTextView = com.kota.telnetUI.textView.TelnetTextViewSmall(context).apply {
            id = R.id.MailBoxPage_ItemView_Author
            layoutParams = LayoutParams(0, LayoutParams.WRAP_CONTENT, 1f).apply {
                marginStart = dp(12f)
            }
            gravity = android.view.Gravity.END
            isSingleLine = true
            setTextColor(getThemeColor(R.attr.bahamut_boardItemAuthorColor))
        }
        metaRow.addView(authorTextView)

        bgView.addView(metaRow)
        row.addView(bgView)

        val verticalDivider = View(context).apply {
            layoutParams = LayoutParams(dp(1f), LayoutParams.MATCH_PARENT)
            setBackgroundColor(getThemeColor(R.attr.bahamut_dividerColor))
        }
        row.addView(verticalDivider)

        val arrowView = com.kota.telnetUI.textView.TelnetTextViewSmall(context).apply {
            id = R.id.ListItem_ArrowView
            layoutParams = LayoutParams(LayoutParams.WRAP_CONTENT, LayoutParams.MATCH_PARENT)
            gravity = android.view.Gravity.CENTER
            setPadding(dp(6f), 0, dp(6f), 0)
            text = getContextString(R.string.arrow_right)
            typeface = android.graphics.Typeface.DEFAULT_BOLD
            setTextColor(getThemeColor(R.attr.bahamut_arrowColor))
            setBackgroundColor(getThemeColor(R.attr.bahamut_arrowBackground))
        }
        row.addView(arrowView)

        addView(row)

        dividerBottom = View(context).apply {
            id = R.id.MailBoxPage_ItemView_DividerBottom
            layoutParams = LayoutParams(LayoutParams.MATCH_PARENT, dp(1f))
            setBackgroundColor(getThemeColor(R.attr.bahamut_dividerColor))
        }
        addView(dividerBottom)
    }

    fun setTitle(title: String?) {
        if (titleTextView != null) {
            titleTextView?.text = Objects.requireNonNullElse<String?>(
                title,
                getContextString(R.string.loading_)
            )
        }
    }

    fun setAuthor(author: String?) {
        if (this@MailBoxPageItemView.authorTextView != null) {
            this@MailBoxPageItemView.authorTextView?.text = Objects.requireNonNullElse<String?>(
                author,
                getContextString(R.string.loading)
            )
        }
    }

    fun setDate(date: String?) {
        if (dateTextView != null) {
            dateTextView?.text = Objects.requireNonNullElse<String?>(
                date,
                getContextString(R.string.loading)
            )
        }
    }

    @SuppressLint("DefaultLocale")
    fun setIndex(number: Int) {
        if (numberTextView != null) {
            if (number > 0) {
                numberTextView?.text = String.format("%1$05d", number)
                return
            }
            numberTextView?.text = getContextString(R.string.loading)
        }
    }

    fun setReply(isReply: Boolean) {
        if (isReply) {
            replyTextView?.visibility = VISIBLE
        } else {
            replyTextView?.visibility = INVISIBLE
        }
    }

    fun setMark(isMarked: Boolean) {
        if (isMarked) {
            markTextView?.visibility = VISIBLE
        } else {
            markTextView?.visibility = INVISIBLE
        }
    }

    fun setRead(isRead: Boolean) {
        // 戰巴哈信件只要看到有沒有讀取
        if (isRead) {
            statusTextView?.text = "◇"
            titleTextView?.setTextColor(getThemeColor(R.attr.bahamut_boardItemNormalReadColor))
            return
        }
        statusTextView?.text = "◆"
        titleTextView?.setTextColor(getThemeColor(R.attr.bahamut_boardItemNormalColor))
    }

    fun clear() {
        setTitle(null)
        setDate(null)
        setAuthor(null)
        setIndex(0)
        setRead(true)
        setReply(false)
        setMark(false)
    }

    fun setDividerBottomVisible(visible: Boolean) {
        if (dividerBottom == null) {
            return
        }
        if (visible) {
            if (dividerBottom?.visibility != VISIBLE) {
                dividerBottom?.visibility = VISIBLE
            }
        } else if (dividerBottom?.visibility != GONE) {
            dividerBottom?.visibility = GONE
        }
    }
}
